ttnn/cpp/ttnn/tensor/layout/tensor_layout.cpp

```diff
@@ -122,33 +159,104 @@ size_t TensorLayout::compute_page_size_bytes(const Size& page_size) const {
     return page_config_.get_page_size_bytes(page_size, dtype_);
 }
 
+Size TensorLayout::compute_physical_shard_shape(const Size& logical_shard_shape) const {
+    // TODO: Alignment before sharding can describe padding along non-height or width dims
+    // However, this is not compatible with logical sharding and we just leave it as nullopt
+    const int alignment_rank = static_cast<int>(alignment_.size());
+    TT_FATAL(alignment_rank <= 2, "Alignment {} must be rank 2 or less to compute physical shard shape", alignment_);
+
+    auto physical_shard_height = CMAKE_UNIQUE_NAMESPACE::round_up(logical_shard_shape.height(), alignment_[-2]);
+    auto physical_shard_width = CMAKE_UNIQUE_NAMESPACE::round_up(logical_shard_shape.width(), alignment_[-1]);
+    return Size{physical_shard_height, physical_shard_width};
+}
+
 Size TensorLayout::compute_physical_shape(const ttnn::SimpleShape& shape) const {
     const int rank = static_cast<int>(shape.rank());
     const int alignment_rank = static_cast<int>(alignment_.size());
-    const int max_rank = std::max(rank, alignment_rank);
+
     size_t width = 1;
     size_t height = 1;
 
-    // Iterate dims in reverse order and ensure alignment
-    // Even tensor of rank 0 or 1 must be aligned (to Tile / Page / Shard)
-    for (int i = -1; i >= -max_rank; --i) {
-        auto& dim = i == -1 ? width : height;
-        if(i >= -rank) {
+    if (memory_config_.shard_spec.has_value() and memory_config_.shard_spec.value().mode == ShardMode::LOGICAL) {
+        // Iterate dims in reverse order
+        for (int i = -1; i >= -rank; --i) {
+            auto& dim = i == -1 ? width : height;
             dim *= shape[i];
         }
 
-        // Align the current dimension if alignment is available
-        if (i >= -alignment_rank) {
-            dim = CMAKE_UNIQUE_NAMESPACE::round_up(dim, alignment_[i]);
+        auto compute_default_logical_shard_shape = [&shape, &rank]() {
+            size_t shard_width = 1;
+            size_t shard_height = 1;
+
+            if (rank >= 1) {
+                shard_width = shape[-1];
+            }
+            if (rank >= 2) {
+                shard_height = shape[-2];
+            }
```

Comment:
The checks are redundant, shape can always handle shape[-1] and shape[-2] with 1 being the default.

Reply:
Code is unused and I removed it